server/src/main/java/org/elasticsearch/gateway/MetaDataStateFormat.java

```diff
@@ -207,20 +225,15 @@ protected Directory newDirectory(Path dir) throws IOException {
         return new SimpleFSDirectory(dir);
     }
 
-    private void cleanupOldFiles(final String prefix, final String currentStateFile, Path[] locations) throws IOException {
-        final DirectoryStream.Filter<Path> filter = entry -> {
-            final String entryFileName = entry.getFileName().toString();
-            return Files.isRegularFile(entry)
-                    && entryFileName.startsWith(prefix) // only state files
-                    && currentStateFile.equals(entryFileName) == false; // keep the current state file around
-        };
-        // now clean up the old files
-        for (Path dataLocation : locations) {
-            logger.trace("cleanupOldFiles: cleaning up {}", dataLocation);
-            try (DirectoryStream<Path> stream = Files.newDirectoryStream(dataLocation.resolve(STATE_DIR_NAME), filter)) {
-                for (Path stateFile : stream) {
-                    Files.deleteIfExists(stateFile);
-                    logger.trace("cleanupOldFiles: cleaned up {}", stateFile);
+    private void cleanupOldFiles(final String currentStateFile, Path[] locations) throws IOException {
+        for (Path location : locations) {
+            Path stateLocation = location.resolve(STATE_DIR_NAME);
+            try (Directory stateDir = newDirectory(stateLocation)) {
+                for (String file : stateDir.listAll()) {
+                    if (file.startsWith(prefix) && file.equals(currentStateFile) == false) {
+                        deleteTempFile(stateLocation, stateDir, file);
+                        logger.trace("cleanupOldFiles: cleaned up {}", stateLocation.resolve(file));
```

Comment:
this is already logged part of `deleteTempFile`?

Reply:
Fixed, also renamed deleteTempFile method, because when used in cleanup it's not temp file, but previous state